openedx/core/djangoapps/credit/api/eligibility.py

```diff
@@ -225,6 +228,15 @@ def set_credit_requirement_status(username, course_key, req_namespace, req_name,
             )
 
     """
+    username = user.username
+    # Check whether user has credit enrollment. Only credit enrollments should be eligible for credit.
+    enrollment_mode, is_active = CourseEnrollment.enrollment_mode_for_user(user, course_key)
+    has_credit_enrollment = (CourseMode.is_credit_slug(enrollment_mode) and is_active)
```

Comment:
I don't think this is the right solution.
The work flow is this:
1. Learner in verified mode passes the credit requirement
2. Learner go to progress page or dashboard and see "Get Credit" button
3. Learner click on "Get Credit" button to checkout credit product
4. After checkout complete successfully, the learner's course_mode is updated from "Verified" to "Credit".
Please note, this `set_credit_requirement_status` is called on step two. At that time, according to system design, the user do not have course_enrollment with "credit" mode. This will always fail even for "Verified" mode learner. With this change, no one can get credit. Let me know if my above workflow or knowledge is wrong.